File: examples/agent-framework-webhooks/python/agent_webhooks/adapters/openai.py
"""OpenAI Agents SDK reply adapter."""

import os
from collections.abc import Awaitable, Callable
from typing import Any

from e2a import AsyncInboundEmail

from agent_webhooks.prompt import email_prompt

INSTRUCTIONS = (
    "Reply helpfully and concisely to the email. Write 1-3 short paragraphs of "
    "body text only; do not include a Subject line or quote the original email."
)

OpenAIRun = Callable[[str], Awaitable[Any]]


class OpenAIReplyAgent:
    def __init__(self, run: OpenAIRun) -> None:
        self._run = run

    @classmethod
    def from_env(cls) -> "OpenAIReplyAgent":
        from agents import Agent, Runner

        agent = Agent(
            name="Email assistant",
            instructions=INSTRUCTIONS,
            model=os.getenv("OPENAI_MODEL", "gpt-5.6"),
        )
        return cls(lambda prompt: Runner.run(agent, prompt))

    async def reply(self, email: AsyncInboundEmail) -> str:
        result = await self._run(email_prompt(email))
        final_output = result.final_output
        return "" if final_output is None else str(final_output)

File: examples/agent-framework-webhooks/typescript/src/adapters/openai.ts
import type { InboundEmail } from "@e2a/sdk/v1";

import type { ReplyAgent } from "../contracts.js";
import { REPLY_INSTRUCTIONS, emailPrompt } from "../prompt.js";

export interface OpenAIResult {
  finalOutput?: unknown;
}

export type OpenAIRun = (prompt: string) => Promise<OpenAIResult>;

export class OpenAIReplyAgent implements ReplyAgent {
  constructor(private readonly run: OpenAIRun) {}

  async reply(email: InboundEmail): Promise<string> {
    const result = await this.run(emailPrompt(email));
    return result.finalOutput == null ? "" : String(result.finalOutput);
  }
}

export interface OpenAISDK {
  Agent: new (options: { name: string; instructions: string; model: string }) => unknown;
  run(agent: unknown, prompt: string): Promise<OpenAIResult>;
}

/** Build the production adapter from the official `@openai/agents` exports. */
export function createOpenAIReplyAgent(
  sdk: OpenAISDK,
  env: Record<string, string | undefined> = process.env,
): OpenAIReplyAgent {
  const agent = new sdk.Agent({
    name: "Email assistant",
    instructions: REPLY_INSTRUCTIONS,
    model: env.OPENAI_MODEL ?? "gpt-5.6",
  });
  return new OpenAIReplyAgent((prompt) => sdk.run(agent, prompt));
}
